### PR TITLE
Update used psabpf library for PTF tests (PSA/eBPF)

### PR DESCRIPTION
**Pull request by tatry**
It adds support, compared to earlier version, for:
- DirectMeter;
- DirectCounter;
- set default action;
- get table entry (limited to single entry for table without implementation).

CC: @osinstom @kmateuszssak

### CODE REVIEW

File: backends/ebpf/tests/ptf/common.py [hunk 1]
```diff
@@ -208,33 +242,75 @@ def table_write(self, method, table, keys, action=0, data=None, priority=None, r
...
+    def table_get(self, table, keys, indirect=False):
+        cmd = "psabpf-ctl table get pipe {} {} ".format(TEST_PIPELINE_ID, table)
+        if indirect:
+            # TODO: cmd = cmd + "ref "
+            self.fail("support for indirect table not supported yet")
+        cmd = cmd + self._table_create_str_from_key(keys=keys)
+        _, stdout, _ = self.exec_ns_cmd(cmd, "Table get entry failed")
+        return json.loads(stdout)[table]
+
+    def table_verify(self, table, keys, action=0, priority=None, data=None, references=None,
```
Comment (mbudiu-vmw) [inline backends/ebpf/tests/ptf/common.py:281]: Some of these functions could use a documentation """

File: backends/ebpf/tests/ptf/common.py [hunk 2]
```diff
@@ -208,33 +242,75 @@ def table_write(self, method, table, keys, action=0, data=None, priority=None, r
...
-            for d in data:
-                cmd = cmd + "{} ".format(d)
+        cmd = cmd + self._table_create_str_from_key(keys=keys)
+        cmd = cmd + self._table_create_str_from_data(data=data, counters=counters, meters=meters)
         if priority:
             cmd = cmd + "priority {}".format(priority)
         self.exec_ns_cmd(cmd, "Table {} failed".format(method))
 
-    def table_add(self, table, keys, action=0, data=None, priority=None, references=None):
+    def table_add(self, table, keys, action=0, data=None, priority=None, references=None,
```
Comment (mbudiu-vmw) [inline backends/ebpf/tests/ptf/common.py:251]: have you considered adding type annotation to your python? 
I find they help a lot with readability.
So actions are encoded as integers? I wonder whether p4info could be used to use action symbolic names.

Comment (tatry) [reply backends/ebpf/tests/ptf/common.py:251]: > have you considered adding type annotation to your python? I find they help a lot with readability.

At some point of time I had an idea to add that annotations, but I abandoned it. I don't remember why. 

> So actions are encoded as integers? I wonder whether p4info could be used to use action symbolic names.

Yes, exactly, actions are encoded as integers. We decided that p4info can't be used at the level of `psabpf[-cli]`, so every think what we can comes from BTF side. DirectCounters and DirectMeters are recognized by name, so for actions it should be also possible, but is not implemented yet.

**Comment by osinstom** [comment]
@mbudiu-vmw can we merge this?

**Review by mbudiu-vmw — APPROVED** [review]

**Pull request merged**